data-prepper-core/src/main/java/com/amazon/dataprepper/pipeline/server/DataPrepperServer.java

```diff
@@ -48,4 +65,42 @@ public void stop() {
         server.stop(0);
         LOG.info("Data Prepper server stopped");
     }
+
+    private HttpServer createHttpServer(final int port) throws IOException {
+        return HttpServer.create(new InetSocketAddress(port), 0);
+    }
+
+    private HttpServer createHttpsServer(final int port,
+                                         final String keystoreFilePath,
+                                         final String passphrase) throws IOException {
+        final SSLContext sslContext;
+
+        try {
+            char[] passphraseArray = passphrase.toCharArray();
+            KeyStore keyStore = KeyStore.getInstance("JKS");
+            keyStore.load(new FileInputStream(keystoreFilePath), passphraseArray);
+
+            KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance("SunX509");
+            keyManagerFactory.init(keyStore, passphraseArray);
+
+            TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance("SunX509");
+            trustManagerFactory.init(keyStore);
+
+            sslContext = SSLContext.getInstance("TLS");
+            sslContext.init(keyManagerFactory.getKeyManagers(), trustManagerFactory.getTrustManagers(), null);
```

Comment:
The source uses `.crt`  and `.key`, would be great if we can follow the same pattern that way customers can have same experience. And JDK has moved towards `pkcs12` from exclusive `jks`.

Reply:
Good point regarding PKCS12 - should be easy to switch to that.

Regarding the `.crt` and `.key` files used by OTelSource/Armeria/Netty - they're actually following PKCS8 and that's accomplished by a lot of heavy lifting in `io.netty.handler.ssl` via [PemReader.java](https://github.com/whg333/netty-4.0.36.Final/blob/master/src/main/java/io/netty/handler/ssl/PemReader.java). 

We can do the same regex matching to strip the comment lines in the private key, loading that into a blank keystore initialized in memory, loading the certificate via CertificateFactory, etc. but it's quite a bit more effort than just loading a keystore file which already contains all that data. (I've been trying to implement the PKCS8 route but haven't quite got it working yet).